src/Chainweb/Pact/PactService.hs

```diff
@@ -746,6 +749,30 @@ validateChainwebTxs cp txValidationTime bh txs doBuyGas
     runValid f (Right r) = f r
     runValid _ l@Left{} = pure l
 
+-- | Validate just the timings of the transaction of a new block. This check is
```

Comment:
I am very confused about how this change has been implemented in `execNewBlock` and vis-a-vis `timingsCheck`. My three complaints are:

- the copy/paste in `newBlockTimingsCheck` from `timingsCheck`
- the fact that `execNewBlock` BOTH calls `timingsCheck` (inside `validateChainwebTxs`) AND `newBlockTimingsCheck` (via this function)
- that there are multiple warnings in the code, new functions like this that dupe noisy stuff like `fmap payloadObj t` etc

Since it seems that the only change you want is to subtract `compatTime` (but please see my comment there asking if we can't also do this in validate, which would eliminate the control flow), then please: just parameterize `timingsCheck` to do the `compatTime` substraction in new block and remove all changes related to this from this file. You can put all the comments about "remove me afterward" there.

Reply:
I tried to implement it in a way that it's easy to delete the code when it's not needed any more after the transition. If both control flows are interleaved it will be harder to delete the transition code without breaking the actual production code.

I'll check if I can reduce the size of the legacy code, without mixing it with non-legacy code.